python/plugins/processing/algs/qgis/Dissolve.py

```diff
@@ -85,37 +85,49 @@ def processAlgorithm(self, progress):
             outFeat.setAttributes(attrs)
             writer.addFeature(outFeat)
         else:
-            unique = vector.getUniqueValues(vlayerA, int(field))
-            nFeat = nFeat * len(unique)
+            fieldIdx = vlayerA.fieldNameIndex(fieldname)
+            unique = vector.getUniqueValues(vlayerA, int(fieldIdx))
+            nFeat = len(unique)
+            myDict = {}
+            attrDict = {}
             for item in unique:
-                first = True
-                add = True
-                features = vector.features(vlayerA)
-                for inFeat in features:
-                    nElement += 1
-                    progress.setPercentage(int(nElement / nFeat * 100))
-                    atMap = inFeat.attributes()
-                    tempItem = atMap[field]
-                    if unicode(tempItem).strip() == unicode(item).strip():
-                        if first:
-                            QgsGeometry(inFeat.geometry())
-                            tmpInGeom = QgsGeometry(inFeat.geometry())
-                            outFeat.setGeometry(tmpInGeom)
-                            first = False
-                            attrs = inFeat.attributes()
-                        else:
-                            tmpInGeom = QgsGeometry(inFeat.geometry())
-                            tmpOutGeom = QgsGeometry(outFeat.geometry())
-                            try:
-                                tmpOutGeom = QgsGeometry(
-                                    tmpOutGeom.combine(tmpInGeom))
-                                outFeat.setGeometry(tmpOutGeom)
-                            except:
-                                raise GeoAlgorithmExecutionException(
-                                    self.tr('Geometry exception while dissolving'))
-                if add:
-                    outFeat.setAttributes(attrs)
-                    writer.addFeature(outFeat)
+                myDict[unicode(item).strip()] = []
+                attrDict[unicode(item).strip()] = None
+
+            unique = None
+
+            for inFeat in features:
+                attrs = inFeat.attributes()
+                tempItem = attrs[fieldIdx]
+                tmpInGeom = QgsGeometry(inFeat.geometry())
+
+                if attrDict[unicode(tempItem).strip()] == None:
```

Comment:
I did not change this, it was already in the code. I reckon it is used because the dissolve field can be any field type. By converting it to strings you can handle them all the same. Do you see any disadvantages in that?